modules/mobile/docs/schemas/CreateAppointmentRequest.yml

```diff
@@ -3,49 +3,68 @@ type: object
 properties:
   kind:
     "$ref": ./AppointmentKind.yml
+    example: 'clinic'
   status:
     "$ref": ./AppointmentStatus.yml
+    example: 'proposed'
   serviceType:
     type: string
-    description: the care type for the appointment
+    description: Used for VA and CC Requests . the care type for the appointment
+    example: 'optometry'
   reasonCode:
-    "$ref": ./CodeableConcept.yml
-  patientIcn:
-    description: The patient ICN
-    type: string
-    nullable: false
+    type: object
+    description: Used for VA Requests. When Other is chosen, coding should be an empty [] and text is populated by comment text.
+    example: {
+      "coding":[
+        {
+          "code":"Routine Follow-up"
+        }
+      ],
+      "text":"Routine Follow-up"
+    }
+    enum:
+      - 'Routine Follow-up'
+      - 'Medication Concern'
+      - 'New Problem'
+      - []
```

Comment:
I don't think this is quite accurate. They all have the brackets. The others just have an object inside the brackets. I'm not sure how to display this, though. Really, the whole object is the option, so maybe it would make more sense to have enums like:
```
- {"coding":[{"code":"Routine Follow-up"}],"text":"Routine Follow-up"}

- {"coding":[{"code":"New Problem"}],"text":"New Problem"}

- {"coding":[{"code":"Medication Concern"}],"text":"Medication Concern"}

- {"coding":[],"text":"comment text box content here"}
```

I'm not sure that's quite right either because the text can vary. I think the most correct thing may be to drill deeper and define the parts of the reasonCode object. Like define that it has a coding array that can be of four enums and a text object that is a string. 

Ultimately, I think this will have to mostly be tribal knowledge that we pass to the FE devs.

Reply:
Yes, that sounds good to me. I went back and drilled into the coding array and text. Just set the enum to the code field and let the text just be a string. put in the description that if coding is empty. it should be the comment box.